zerver/lib/bugdown/__init__.py

```diff
@@ -1587,12 +1588,23 @@ class RealmFilterPattern(markdown.inlinepatterns.Pattern):
     def __init__(self, source_pattern: str,
                  format_string: str,
                  markdown_instance: Optional[markdown.Markdown]=None) -> None:
+        self.source_pattern = source_pattern
         self.pattern = prepare_realm_pattern(source_pattern)
-        self.format_string = format_string
+        all_filters = RealmFilter.objects.filter(pattern=self.source_pattern)
+        self.data = {obj.stream_id: obj.url_format_string for obj in all_filters}
         markdown.inlinepatterns.Pattern.__init__(self, self.pattern, markdown_instance)
 
     def handleMatch(self, m: Match[str]) -> Union[Element, str]:
         db_data = self.markdown.zulip_db_data
+        msg = self.markdown.zulip_message
+        current_stream = msg.recipient.type_id
+        filter_list_streams = self.data.keys()
+        if current_stream in filter_list_streams:
```

Comment:
I don't understand this if-else sequence. Why is it not as simple as:

```
if self.stream and not current_stream == self.stream:
    return None
process()
```

I'll leave further comments in CZO.